.github/workflows/mariadb.yml

```
@@ -0,0 +1,67 @@
+name: Build Mroonga with MariaDB main
+on:
+  push:
+  pull_request:
+    paths:
+      - "**/*.cpp"
+      - "**/*.h"
+      - "**/*.hpp"
+      - "**/CMakeLists.txt"
+      - "**/Makefile.am"
+      - ".github/workflows/mariadb.yml"
+      - "configure.ac"
+      - "doc/**"
+      - "mysql-test/**"
+      - "packages/**"
+      - "version_*"
+  schedule:
+    - cron: |
+        0 0 * * *
+concurrency:
+  group: ${{ github.head_ref || github.sha }}-${{ github.workflow }}
+  cancel-in-progress: true
+jobs:
+  build:
+    name: Build with MariaDB main
+    timeout-minutes: 30
+    strategy:
+      fail-fast: false
+    runs-on: ubuntu-24.04
```

もしあえて固定する必要がなかったら、`ubuntu-latest`でUbuntu 24.04を利用してくれるので、常に最新を使う方針がおすすめです。
```suggestion
    runs-on: ubuntu-latest
```